### PR TITLE
net: shell: Display TWT response details

### PR DESCRIPTION
**Pull request by ajayparida**
Show TWT response parameters details.

### CODE REVIEW

**Comment by ajayparida** [comment]
@sachinthegreen @krish2718 @rado17

**Comment by ajayparida** [comment]
@sachinthegreen @krish2718 @rado17

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -157,14 +157,26 @@ static void handle_wifi_twt_event(struct net_mgmt_event_callback *cb)
...
 
-		/* If accepted, then no need to print TWT params */
-		if (resp->setup_cmd != WIFI_TWT_SETUP_CMD_ACCEPT) {
-			print(context.sh, SHELL_NORMAL,
-			      "TWT parameters: trigger: %s wake_interval: %d us, interval: %lld us\n",
-			      resp->setup.trigger ? "trigger" : "no_trigger",
-			      resp->setup.twt_wake_interval,
-			      resp->setup.twt_interval);
-		}
+		print(context.sh, SHELL_NORMAL, "negotiation_type: %s\n",
```
Comment (krish2718) [inline subsys/net/l2/wifi/wifi_shell.c:160]: We already are dumping TWT info https://github.com/zephyrproject-rtos/zephyr/blob/ca330e04e72a8330b8f48badb02048483c98c82d/subsys/net/l2/wifi/wifi_shell.c#L478 so, better to move that to a helper function and use it here.

File: subsys/net/l2/wifi/wifi_shell.c [hunk 2]
```diff
@@ -148,23 +148,50 @@ static void handle_wifi_disconnect_result(struct net_mgmt_event_callback *cb)
 	}
 }
 
+static void print_twt_params(uint8_t dialog_token, uint8_t flow_id,
```
Comment (krish2718) [inline subsys/net/l2/wifi/wifi_shell.c:151]: isn't it easier and scalable just to pass the structure instead of individual parameters?

Comment (ajayparida) [reply subsys/net/l2/wifi/wifi_shell.c:151]: The twt setup command(wifi_twt_flow_info) and twt response (wifi_twt_params) does not have uniform strructure. In that case we have to create local structure of  wifi_twt_flow_info type,assign each parameter and then pass it to helper function. Is there any other way?

Comment (krish2718) [reply subsys/net/l2/wifi/wifi_shell.c:151]: Both setup and response use same structure `struct wifi_twt_params`, I guess you mean twt setup/response vs. flow?

Looking at the structures we can easily use `struct wifi_twt_flow_info` inside `struct wifi_twt_params`, and then have helpers to print out request/response and flow information?

Comment (ajayparida) [reply subsys/net/l2/wifi/wifi_shell.c:151]: Yeah, but needs a lot of changes to existing code base(both zephyr and nrf). 
I have no problem in doing so, if it has to be that way, the requirement was to show the parameters.

Comment (krish2718) [reply subsys/net/l2/wifi/wifi_shell.c:151]: Its fine we can do the refactoring later if its too much trouble.

**Review by krish2718 — DISMISSED** [review]

**Review by rlubos — APPROVED** [review]

File: subsys/net/l2/wifi/wifi_shell.c [hunk 3]
```diff
@@ -148,23 +148,50 @@ static void handle_wifi_disconnect_result(struct net_mgmt_event_callback *cb)
...
+		      implicit ? "true" : "false");
+	shell_fprintf(context.sh, SHELL_NORMAL, "TWT announce: %s\n",
+		      announce ? "true" : "false");
+	shell_fprintf(context.sh, SHELL_NORMAL, "TWT trigger: %s\n",
+		      trigger ? "true" : "false");
+	shell_fprintf(context.sh, SHELL_NORMAL, "TWT wake interval: %d us\n",
+		      twt_wake_interval);
+	shell_fprintf(context.sh, SHELL_NORMAL, "TWT interval: %lld us\n",
+		      twt_interval);
+	shell_fprintf(sh, SHELL_NORMAL, "========================\n");
```
Comment (rlubos) [inline subsys/net/l2/wifi/wifi_shell.c:175]: This should be `context.sh`, please fix.

**Review by rlubos — CHANGES_REQUESTED** [review]

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**